### PR TITLE
fix: next/prev/moveTo not work after calling disableInput while holding the panel

### LINKED ISSUE #407: Methods won't work after calling `disableInput` while holding the panel
## Description
Methods won't work after calling `disableInput` while holding the panel.
This is because the internal state is not changing after calling `disableInput`.

## Steps to check or reproduce
Call disableInput on touchstart, then call next/prev/moveTo after.

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#407 

## Details
This fixes #407

### CODE REVIEW

File: src/components/Viewport.ts [hunk 1]
```diff
@@ -674,7 +689,7 @@ export default class Viewport {
     }
 
     this.axes.destroy();
-    this.panInput.destroy();
+    this.panInput && this.panInput.destroy();
```
Comment (mixed) [inline src/components/Viewport.ts:692]: nit
```diff
- this.panInput && this.panInput.destroy();
+ this.panInput!.destroy();
```

**Review by mixed — APPROVED** [review]

File: src/components/Viewport.ts (hunk 1 above)
Comment (WoodNeck) [reply src/components/Viewport.ts:692]: I thought Flicking was still using Typescript v2.8, I'll fix that.

File: src/components/Viewport.ts [hunk 2]
```diff
@@ -384,11 +384,26 @@ export default class Viewport {
...
+        scale: options.horizontal ? [-1, 0] : [0, -1],
+      });
+
+      this.axes.connect(options.horizontal ? ["flick", ""] : ["", "flick"], this.panInput);
+    }
   }
 
   public disable(): void {
-    this.panInput.disable();
+    if (this.panInput) {
```
Comment (daybrush) [inline src/components/Viewport.ts:401]: @WoodNeck 

It probably overlaps with the init code.

File: src/components/Viewport.ts [hunk 3]
```diff
@@ -384,11 +384,26 @@ export default class Viewport {
...
+      this.panInput.destroy();
+      this.panInput = null;
```
Comment (daybrush) [inline src/components/Viewport.ts:403]: Is the original paninput disconnected?

File: src/components/Viewport.ts [hunk 4]
```diff
@@ -384,11 +384,26 @@ export default class Viewport {
...
+    if (!this.panInput) {
+      const options = this.options;
+      this.panInput = new PanInput(this.viewportElement, {
+        inputType: options.inputType,
+        thresholdAngle: options.thresholdAngle,
+        iOSEdgeSwipeThreshold: options.iOSEdgeSwipeThreshold,
+        scale: options.horizontal ? [-1, 0] : [0, -1],
+      });
+
+      this.axes.connect(options.horizontal ? ["flick", ""] : ["", "flick"], this.panInput);
```
Comment (daybrush) [inline src/components/Viewport.ts:396]: @WoodNeck

It probably overlaps with the init code.

File: src/components/Viewport.ts (hunk 3 above)
Comment (WoodNeck) [reply src/components/Viewport.ts:403]: Yes it calls `disconnect()` on destroy.
image

Comment (daybrush) [reply src/components/Viewport.ts:403]: @WoodNeck 

I mean, it is said to `axes.disconnect`

Comment (WoodNeck) [reply src/components/Viewport.ts:403]: Looks like `axes.disconnect` also calls `disconnect` of a inputType, so I guess it's okay?
image

File: src/components/Viewport.ts (hunk 4 above)
Comment (WoodNeck) [reply src/components/Viewport.ts:396]: Thanks for checking, I've updated my code

File: src/components/Viewport.ts (hunk 3 above)
Comment (daybrush) [reply src/components/Viewport.ts:395]: Um... it needs to call the destroy method of paninput and then the disconnect method.

Comment (WoodNeck) [reply src/components/Viewport.ts:395]: Why should it have to be?

Comment (daybrush) [reply src/components/Viewport.ts:395]: Well. Should the axes be connected, shouldn't I disconnect? Even if the axes don't call disconnect, there seems to be no problem.

**Review by daybrush — APPROVED** [review]
LGTM

File: src/components/Viewport.ts (hunk 3 above)
Comment (WoodNeck) [reply src/components/Viewport.ts:395]: I mean, as the PanInput calls `disconnect()` while calling `destroy()` why should it have to be called again?

**Pull request merged**